geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/PersistentBucketRecoverer.java

```diff
@@ -197,17 +209,23 @@ public void run2() {
       // Log and bail
       logger.error(e.getMessage(), e);
     } finally {
-      /*
-       * Our job is done. Stop listening to the bucket advisors.
-       */
-      removeListeners();
-      /*
-       * Make sure the recovery completion message was printed to the log.
-       */
-      for (RegionStatus region : regions) {
-        if (!region.loggedDoneMessage) {
-          region.logDoneMessage();
+      if (!regions.isEmpty()) {
```

Comment:
I'm not sure this if check is necessary. Everything inside here will work if regions is empty, and will have no effect if regions is not.